net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+
+    # source
+    move ${worksrcpath} \
+        ${destroot}${prefix}/src/${ccsname}
+
+    # calenderserver configuration
+    xinstall -m 0600 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/calendarserver.plist \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/Config/calendarserver.plist.macports
+
+    # nginx reverse proxy
+    xinstall -m 0640 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/nginx.conf \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx.conf.macports
+    xinstall -m 0644 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/default.html \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx_root
+
+    # ccs-calendarserver's postgres code points to `PSQL = "../postgresql/_root/bin/psql"`
+    ln -s ${prefix} \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/postgresql/_root
+}
+
+# Network configuration
+# hard-coded examples
+set host                host
+set domain              domain
+set tld                 tld
+set fullhost            ${host}.${domain}.${tld}
+set domaintld           ${domain}.${tld}
+set tls_certificate_name \
+                        ${fullhost}
+
+startupitem.create     yes
+
+startupitems \
+    name               calendarserver \
+    executable \
+        ${calendarserverdir}/Library/CalendarServer/${ccsname}/bin/caldavd
+
+# use defaults, edit launch daemon plists in post-activate below
+# executable ${prefix}/sbin/nginx -g "daemon off;" -p ${calendarserverdir}/Library/CalendarServer/logs -c ${calendarserverdir}/Library/CalendarServer/etc/nginx.conf
+# pidfile auto ${calendarserverdir}/Library/CalendarServer/logs/nginx.pid
+startupitems-append \
+    name               calendarserver_proxy \
+    executable         ${prefix}/sbin/nginx \
+    pidfile            auto
+
+post-activate {
+    # personal data outside of ${prefix}
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer] } {
+        xinstall -m 0755 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer
+    }
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer/Data] } {
+        xinstall -m 0700 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer/Data
+    }
+
+    # delete previous install (see destroot notes above)
+    foreach f_or_d {
+        bin
+        environment.sh
+        lib
+        roots
+        virtualenv
+        } {
+        if { [file exists \
+                  ${calendarserverdir}/Library/CalendarServer/${ccsname}/${f_or_d}]
+            } {
+            delete \
+                ${calendarserverdir}/Library/CalendarServer/${ccsname}/${f_or_d}
+        }
+    }
+
+    # macOS-native Python needs the pip and virtualenv modules
+    if { ![file executable /usr/local/bin/pip] } {
+        system "/usr/bin/easy_install pip"
```

Comment:
I don't consider this to be an acceptable practice for ports.

Why can't you use python and packages from MacPorts?

Reply:
Tried many times, many ways. ccs-calendarserver won’t build without native Python.

It builds and works great using native Python, so I’m using it.

I wouldn’t forego having a calendar server just because it’s built using macOS native Python rather than macports.

There are much bigger looming issues—Python 2.7 is EOL in four months.

Reply:
Finally got MacPorts python working with this install. Native `easy_install` is gone now.